### PR TITLE
Update magellan and multifile test

### PR DESCRIPTION
**Pull request by Maciek416**
This PR:
  - :white_check_mark: Updates the boilerplate to have multiple test specs.
  - :art: Adds a common `base` module where the adapter and mock server are started.
  - :memo: Adds some documentation around how `magellan` supplies a mocking port, and when/when not to use it.
  - :arrow_up: Updates `magellan` to `2.0.2`

cc @chaseadamsio

### CODE REVIEW

**Comment by ryan-roemer** [comment]
Nice!

File: spec/base.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+after(function (done) {
+  if (!server) { return done(); }
+  server.close(done);
+});
+
+module.exports = {
+  adapter: adapter,
+  host: HOST
+}
```
Comment (chaseadamsio) [inline spec/base.js]: nit picking, but to do you normally prefer a semi-colon to close out objects?

Comment (Maciek416) [inline spec/base.js]: @chaseadamsio yep. This was a mistake

Comment (ryan-roemer) [inline spec/base.js]: @chaseadamsio -- Good eye!

We should lint the test files ;)

Comment (chaseadamsio) [inline spec/base.js]: `lint` all the things.

**Pull request merged**